### PR TITLE
Use overlayfs

### PR DESCRIPTION
**Pull request by jml**

### CODE REVIEW

File: roles/docker/files/10-docker.conf [hunk 1]
```diff
@@ -1,3 +1,5 @@
 [Service]
 ExecStart=
-ExecStart=/usr/bin/dockerd --log-opt max-size=100m --log-opt max-file=10  -H fd:// --log-level=warn --bridge=none --iptables=false --ip-masq=false -g /mnt/containers/docker
+# For storage-driver=overlay, see
+# https://github.com/weaveworks/service-conf/issues/593 for context.
```
Comment (tomwilkie) [inline roles/docker/files/10-docker.conf]: I wouldn't include a link to a private repo in a public one.

Comment (jml) [inline roles/docker/files/10-docker.conf]: Fixed. (Not really fixed). 

What would you suggest doing instead?

**Comment by tomwilkie** [comment]
You stuck a link on the private issue, which is what I would have done.

You could also say why (kernel deadlocks in aufs), without linking to the
internal issue.

I'm not super fussed though, enough references to this repo have leaked
out, and it's existence is not a secret.

**Comment by jml** [comment]
@tomwilkie OK to merge?

**Comment by tomwilkie** [comment]
Sure, if you want!  I wouldn't merge yet, until you've tested it works on a single machine - this will cause an alert on prod when merged (as config will differ).

**Pull request merged**